core/domain/exp_domain.py

```diff
@@ -2308,8 +2382,31 @@ def from_yaml(cls, exploration_id, yaml_content):
                 outside the range [EARLIEST_SUPPORTED_EXP_SCHEMA_VERSION,
                 CURRENT_EXP_SCHEMA_VERSION].
         """
-        exploration_dict = cls._migrate_to_latest_yaml_version(yaml_content)
+        exploration_dict = cls._migrate_to_latest_yaml_version(
+            yaml_content, exploration_id)
         exploration_dict['id'] = exploration_id
+
+        exploration = cls(
+            exploration_dict['id'],
+            exploration_dict['title'],
+            exploration_dict['category'],
+            exploration_dict['objective'],
+            exploration_dict['language_code'],
+            exploration_dict['tags'],
+            exploration_dict['blurb'],
+            exploration_dict['author_notes'],
+            exploration_dict['states_schema_version'],
+            exploration_dict['init_state_name'],
+            exploration_dict['states'],
+            exploration_dict['param_specs'],
+            exploration_dict['param_changes'],
+            exploration_dict['schema_version'],
+            exploration_dict['auto_tts_enabled'],
+            exploration_dict['correctness_feedback_enabled'])
+
+        exploration_dict['proto_size_in_bytes'] = (
```

Comment:
Create a new function `add_id_and_proto_size_in_bytes_to_dict` as this part is getting used at two places.

Reply:
The calculation is happening at init only `self.proto_size_in_bytes = self.get_proto_size()`. But in order to input the proto size to the exploration_dict, we need to pass all the data in the form 
```py
cls(....)
```
This will give back us an exploration domain object from which we can extract the proto size, and this way all the exploration_dict can migrate from this function.